packages/nx-container/src/generators/init/generator.ts

```diff
@@ -42,6 +43,9 @@ export default async function (tree: Tree, options: InitGeneratorSchema) {
             ],
           },
         },
+        configurations: {
+          production: {},
+        },
```

Comment:
The dependsOn is enough, please remove the `configurations` property to avoid confusions with the options property

Reply:
> The dependsOn is enough

With `dependsOn` only the `container` task will execute `<project>:build`.

The only way I have found to force the `<project>:build:production` to be executed is to also add a that `production` config. Then you can force using the `build:production` by passing `--configuration=production`.

Reply:
At least with latest version of nx, only with `dependsOn` and passing `--prod` or `--configuration=production`, runs `build:production`... Can you check again please?
<img width="497" alt="Screenshot 2023-01-07 at 21 52 07" src="https://user-images.githubusercontent.com/371939/211175760-aaa5fc99-cfba-4c1c-940b-bc2d2a5b3372.png">

Reply:
Oh you are right...

I was basing my comments on the Nx Console:

![image](https://user-images.githubusercontent.com/248818/211176095-8d5d9854-e6e1-4395-8eab-1d6fa2c82ed5.png)

See how `build` shows `production` and `container` not.

I'll report that to Nx and will update the PR.